docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastForest.cs

```diff
@@ -0,0 +1,95 @@
+
+            // Convert the examples list to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(examples);
+
+            // Define the trainer.
+            var pipeline = mlContext.Regression.Trainers.FastForest();
+
+            // Train the model.
+            var model = pipeline.Fit(trainingData);
+
+            // Create testing examples. Use different random seed to make it different from training data.
+            var testData = mlContext.Data.LoadFromEnumerable(GenerateRandomDataPoints(500, seed:123));
+
+            // Run the model on test data set.
+            var transformedTestData = model.Transform(testData);
+
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
+
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(5))
+                Console.WriteLine($"Label: {p.Label:F3}, Prediction: {p.Score:F3}");
+
+            // Expected output:
+            //   Label: 0.985, Prediction: 0.864
+            //   Label: 0.155, Prediction: 0.164
+            //   Label: 0.515, Prediction: 0.470
+            //   Label: 0.566, Prediction: 0.501
+            //   Label: 0.096, Prediction: 0.138
+
+            // Evaluate the overall metrics
+            var metrics = mlContext.Regression.Evaluate(transformedTestData);
+            SamplesUtils.ConsoleUtils.PrintMetrics(metrics);
+
+            // Expected output:
+            //   Mean Absolute Error: 0.06
+            //   Mean Squared Error: 0.01
+            //   Root Mean Squared Error: 0.07
+            //   RSquared: 0.93
+        }
+
+        private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count, int seed=0)
+        {
+            var random = new Random(seed);
+            float randomFloat() => (float)random.NextDouble();
+            for (int i = 0; i < count; i++)
+            {
+                var label = randomFloat();
+                yield return new DataPoint
+                {
+                    Label = label,
+                    // Create random features that are correlated with label.
+                    Features = Enumerable.Repeat(label, 50).Select(x => x + randomFloat()).ToArray()
+                };
+            }
+        }
+
+        // Example with label and 50 feature values. A data set is a collection of such examples.
+        private class DataPoint
+        {
+            public float Label { get; set; }
+            [VectorType(50)]
+            public float[] Features { get; set; }
+        }
+
+        // Class used to capture predictions.
+        private class Prediction
+        {
+            // Original label.
+            public float Label { get; set; }
+            // Predicted score from the trainer.
+            public float Score { get; set; }
+        }
```

Comment:
This is going to be common piece of code in all the example (I assume). Did we think of templating it in some way so that in future if need to change it we can change it at one place?

Reply:
we're adding boilerplate code in order to make the samples self-contained without dependency on external files that users have to look up. please see #2726. 

---
In reply to: [267029777](https://github.com/dotnet/machinelearning/pull/2999#discussion_r267029777) [](ancestors = 267029777)

Reply:
I mean this: [https://github.com/dotnet/machinelearning/pull/3017](https://github.com/dotnet/machinelearning/pull/3017)
Obviously not going to be in this PR.

---
In reply to: [267038999](https://github.com/dotnet/machinelearning/pull/2999#discussion_r267038999) [](ancestors = 267038999,267029777)